dockstore-webservice/src/main/java/io/dockstore/webservice/helpers/GitHubSourceCodeRepo.java

```
@@ -394,24 +403,25 @@ public Workflow setupWorkflowVersions(String repositoryId, Workflow workflow, Op
 
         // when getting a full workflow, look for versions and check each version for valid workflows
         List<Triple<String, Date, String>> references = new ArrayList<>();
+
+        GHRef[] refs = {};
         try {
-            GHRef[] refs = repository.getRefs();
-            for (GHRef ref : refs) {
-                Triple<String, Date, String> referenceTriple = getRef(ref, repository);
-                if (referenceTriple != null) {
-                    if (versionName.isEmpty() || Objects.equals(versionName.get(), referenceTriple.getLeft())) {
-                        references.add(referenceTriple);
-                    }
+            refs = getBranchesAndTags(repository);
+        } catch (IOException ex) {
```

Same as above, I would let it throw the way it did before.